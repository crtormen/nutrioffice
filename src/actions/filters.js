// SET_TEXT_FILTER Action Generator
export const setTextFilter = ( text = '') => ({
    type: 'SET_TEXT_FILTER',
    text
});

// SORT_BY_NAME Action Generator
export const sortByName = () => ({
    type: 'SORT_BY_NAME'
});

// // SORT_BY_DATE Action Generator
// export const sortByDate = () => ({
//     type: 'SORT_BY_DATE'
// });

// //SET_START_DATE
// export const setStartDate = ( startDate ) => ({
//     type: 'SET_START_DATE',
//     startDate
// });

// //SET_END_DATE
// export const setEndDate = ( endDate ) => ({
//     type: 'SET_END_DATE',
//     endDate
// });
