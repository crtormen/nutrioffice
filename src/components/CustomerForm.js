import React from 'react';


export default class CustomerForm extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            name: props.customer ? props.customer.name : '',
            email: props.customer ? props.customer.email : '',
            phone: props.customer ? props.customer.phone : '',
            born: props.customer ? props.customer.born : '',
            occupation: props.customer ? props.customer.occupation : '',
            instagram: props.customer ? props.customer.instagram : '',
           // calendarFocused: false,
            error: ''
        };
    }


    onSubmit = (e) => {
        e.preventDefault();

        if (!this.state.name || !this.state.email || !this.state.phone) {
            this.setState(() => ({ error: "VocÃª deve informar seu nome, email e celular."})); 
        } else {
            this.setState(() => ({ error: ""})); 
            // Runs parent method
            this.props.onSubmit({
                name: this.state.name,
                email: this.state.email,
                phone: this.state.phone,
                born: this.state.born,
                occupation: this.state.occupation,
                instagram: this.state.instagram                 
            });
        };

    }

    onNameChange = (e) => {
        const name = e.target.value;
        this.setState(() => ({ name }));
    };

    onEmailChange = (e) => {
        const email = e.target.value;
        this.setState(() => ({ email }));
    };

    onPhoneChange = (e) => {
        const phone = e.target.value;
        this.setState(() => ({ phone }));
    };

    onOccupationChange = (e) => {
        const occupation = e.target.value;
        this.setState(() => ({ occupation }));
    };

    onBornChange = (e) => {
        const born = e.target.value;
        this.setState(() => ({ born }));
    };

    onInstagramChange = (e) => {
        const instagram = e.target.value;
        this.setState(() => ({ instagram }));
    };

    render() {
        return (
            <div>
            {this.state.error && <p>{this.state.error}</p>}
            <form onSubmit={this.onSubmit}>
                    <input 
                        type="text" 
                        placeholder="Nome completo" 
                        autoFocus 
                        value={this.state.name}
                        onChange={this.onNameChange}
                    />
                    <input 
                        type="text" 
                        placeholder="Data de nascimento" 
                        value={this.state.born}
                        onChange={this.onBornChange}
                    />
                    <input 
                        type="text" 
                        placeholder="Email"
                        value={this.state.email}
                        onChange={this.onEmailChange}
                    />
                    <input 
                        type="text" 
                        placeholder="Celular" 
                        value={this.state.phone}
                        onChange={this.onPhoneChange}
                    />
                    <input 
                        type="text" 
                        placeholder="ProfissÃ£o" 
                        value={this.state.occupation}
                        onChange={this.onOccupationChange}
                    />
                    <input 
                        type="text" 
                        placeholder="instagram" 
                        value={this.state.instagram}
                        onChange={this.onInstagramChange}
                    />
                <button>Salvar Dados</button>
            </form>
            </div>
        );
    }
}
