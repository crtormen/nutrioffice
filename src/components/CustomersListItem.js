import React from 'react';
import { Link } from 'react-router-dom';

const CustomersListItem = ({id, name, email}) => (
    <div>
        <h3><Link to={`edit/${id}`}>{name}</Link> - {email}</h3>    
    </div>
);

export default CustomersListItem;