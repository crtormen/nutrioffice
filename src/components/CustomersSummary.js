import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import selectCustomers from '../selectors/customers';

export const CustomersSummary = ({ customerCount }) => {
    const customerWord = customerCount === 1 ? 'paciente' : 'pacientes';
    return (
        <div>
            <div>
                <h2>Mostrando <span>{ customerCount }</span> {customerWord} </h2>
                <div>
                    <Link to="/create">+ Novo Paciente</Link>
                </div>
            </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleCustomers = selectCustomers(state.customers, state.filters);
    return {
        customerCount: visibleCustomers.length
    }
}

export default connect(mapStateToProps)(CustomersSummary);