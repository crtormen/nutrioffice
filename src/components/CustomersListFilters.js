import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByName } from '../actions/filters';

export class CustomersListFilters extends React.Component {
    
    onTextChange = (e) => {
        this.props.setTextFilter(e.target.value);
    }

    render() {
        return (
            <div>
                <input 
                    type="text" 
                    value={this.props.filters.text} 
                    onChange={this.onTextChange} 
                    placeholder="Busca..."
                />
            </div>
        );
    };
}

const mapDispatchToProps = (dispatch) => ({
    setTextFilter: (text) => dispatch(setTextFilter(text)),
    sortByName: () => dispatch(sortByName())
});

const mapStateToProps = (state) => ({
    filters: state.filters
});

export default connect(mapStateToProps, mapDispatchToProps)(CustomersListFilters);