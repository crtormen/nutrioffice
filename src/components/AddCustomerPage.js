import React from 'react';
import { connect } from 'react-redux';
import { startAddCustomer } from '../actions/customers';
import CustomerForm from './CustomerForm';

export class AddCustomerPage extends React.Component {

    onSubmit = (customer) => {
        this.props.startAddCustomer(customer);
        this.props.history.push('/dashboard');
    }

    render() {
        return (
            <div>
                <h2>Novo Paciente</h2>
                <CustomerForm onSubmit={this.onSubmit} />
            </div>
        )
    }
}

const mapDispatchToProps = (dispatch, props) => ({
    startAddCustomer: (customer) => dispatch(startAddCustomer(customer))
});

export default connect(undefined, mapDispatchToProps)(AddCustomerPage);
