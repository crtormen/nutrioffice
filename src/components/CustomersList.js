import React from 'react';
import { connect } from 'react-redux';
import selectCustomers from '../selectors/customers';
import CustomersListItem from './CustomersListItem';

export const CustomersList = (props) => (
    <div>
    <h2>Lista de Pacientes</h2>
    {
        props.customers.length === 0 ? (
            <p>Nenhum paciente encontrado.</p>
        ) : (
            props.customers.map((customer) => {
                return (
                    <CustomersListItem 
                        key={customer.id}
                        {...customer}
                    />
                )
            })
        )
    }
    </div>
);

const mapStateToProps = (state) => ({
    customers: selectCustomers(state.customers, state.filters)
});

//connect to store to receive and show always the most updated list of customers
export default connect(mapStateToProps)(CustomersList);