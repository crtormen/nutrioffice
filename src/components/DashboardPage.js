import React from 'react';
import CustomersList from './CustomersList';
import CustomersListFilters from './CustomersListFilters';
import CustomersSummary from './CustomersSummary';

export const DashboardPage = () => (
    <div>
        <CustomersSummary />
        <CustomersListFilters />
        <CustomersList />        
    </div>
);

export default DashboardPage;