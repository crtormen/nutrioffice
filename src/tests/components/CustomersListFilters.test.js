import React from 'react';
import { shallow } from 'enzyme';
import { CustomersListFilters } from '../../components/CustomersListFilters';
import { filters, altFilters } from '../fixtures/filters';

let setTextFilter, sortByName, wrapper;

beforeEach(() => {
    setTextFilter = jest.fn();
    sortByName = jest.fn();
    wrapper = shallow(
        <CustomersListFilters 
            filters={filters} 
            setTextFilter={setTextFilter}
            sortByName={sortByName}
        />
    );
});

test("should render CustomersListFilters correctly", () => {
    expect(wrapper).toMatchSnapshot();
});

test("should handle text change", () => {
    const value = "Bart";
    wrapper.find('input').simulate('change',{
        target: { value }
    });

    expect(setTextFilter).lastCalledWith(value);
});

// test('should sort by name', () => {
//     wrapper.setProps({
//         filters: altFilters
//     });
//     wrapper.find('select').simulate('change', {
//         target: { value: 'date' }
//     } )
//     expect(sortByDate).toBeCalled();
// });
