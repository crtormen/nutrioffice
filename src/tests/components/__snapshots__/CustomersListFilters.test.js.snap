// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CustomersListFilters correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    placeholder="Busca..."
    type="text"
    value=""
  />
</div>
`;
